### DISCUSSION TITLE
Can I use "from osgeo import gdal" with this lambda layer?

### DISCUSSION THREAD

**Discussion by mirouml** [lambgeo/docker-lambda#79] [Q&A] [answered]
Thank you very much for this repository and all the hard work put into it. It made it really easy for me to set up lambda with layer as total rookie in this space. In my case:
1. create lambda using python 3.11
2. Add layer using "Specify an ARN" - "arn:aws:lambda:ap-southeast-2:524387336408:layer:gdal38:4"
3. Under Configuration add 2 environment variables:
   - GDAL_DATA	/var/task/share/gdal
   - PROJ_LIB	/var/task/share/proj

And that was it. Now I can run e.g.:
```python
result = subprocess.run(
    ['ogrinfo', '-ro', '-al', '-so', local_geojson_filename],
    check=True,
    stdout=subprocess.PIPE,
    stderr=subprocess.PIPE,
    text=True
)
```

While this works great I would much rather use:
```python
from osgeo import gdal
```
But this fails. Is there a way to use it with the setup mentioned above? Am I missing something, doing something wrong or is this lambda layer not meant to provide this functionality? Any guidance on this appreciated.

**Comment by vincentsarago** [comment, marked as answer]
you need to create your own package as show in https://github.com/lambgeo/docker-lambda?tab=readme-ov-file#2-advanced-need-other-dependencies

**Reply by mirouml** [reply]
Thank you. It took me a while to understand how to do all this with Docker on Windows but got there at the end. I have decided to go with creating additional layer to work with provided GDAL layer instead of creating package.zip. That way it allows me to do quick edits to my prototype code in AWS console in code source, deploy and test.
